File: ios/Plugin/ClipboardPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(ClipboardPlugin, "Clipboard",
           CAP_PLUGIN_METHOD(read, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(readURL, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(readImage, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(write, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(writeURL, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(writeImage, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(clear, CAPPluginReturnPromise);
)

File: ios/Plugin/ClipboardPlugin.swift
import Foundation
import Capacitor

@objc(ClipboardPlugin)
public class ClipboardPlugin: CAPPlugin {
    
    private let null = NSNull()
    
    @objc func read(_ call: CAPPluginCall) {
        call.unimplemented("Not yet implemented!")
    }

    @objc func readURL(_ call: CAPPluginCall) {
        let url = UIPasteboard.general.hasURLs ? UIPasteboard.general.url : nil;
        call.resolve([
            "url": url?.absoluteString ?? null
        ])
    }

    @objc func readImage(_ call: CAPPluginCall) {
        call.unimplemented("Not yet implemented!")
    }
    
    @objc func write(_ call: CAPPluginCall) {
        call.unimplemented("Not yet implemented!")
    }

    @objc func writeURL(_ call: CAPPluginCall) {
        call.unimplemented("Not yet implemented!")
    }

    @objc func writeImage(_ call: CAPPluginCall) {
        call.unimplemented("Not yet implemented!")
    }
    
    @objc func clear(_ call: CAPPluginCall) {
        UIPasteboard.general.items = []
        call.resolve()
    }

}

